src/components/DatePicker/DatePicker.jsx

```diff
@@ -145,12 +145,22 @@ export const DatePicker = ({
                 const to = toDate(now(day, timezoneName).endOf("day"), false);
 
                 onChange({ from, to }, options, event);
-            } else {
+            } else if (DateUtils.isDayBefore(value.from, toDate(now(day, timezoneName)))) {
                 onChange(
```

Comment:
Can you add a one line comment for this condition. Same as what you've done below.